cmake/toolchain.cmake

```diff
@@ -1,6 +1,46 @@
 # Copyright (c) 2020 ARM Limited. All rights reserved.
 # SPDX-License-Identifier: Apache-2.0
 
+# Set the system processor depending on the CPU core type
+if (MBED_CPU_CORE STREQUAL Cortex-A9.cmake)
+    set(CMAKE_SYSTEM_PROCESSOR cortex-a9)
+elseif (MBED_CPU_CORE STREQUAL Cortex-M0+)
```

Comment:
What is the benefit having these here but not in each core cmake ? we set there defines and other related core things, so why not also `CMAKE_SYSTEM_PROCESSOR ` ?

Reply:
It is to remove the toolchain dependency on the core modules. It is the toolchain (precisely ARM toolchain) that needs it defined when setting up the toolchain. When it is in the core module we create a dependency on that module to be included before this.

Reply:
I recall now when I was reviewing toolchain files in CMake repository, they check it there . But that does not mean it belongs to toolchain, its internal detail of CMake. If we look from a user perspective, I would expect to have these set in core (if there is core error, I would look for core settings not toolchain).

Reply:
When this variable is not set you get the following error: 
```
CMake Error at /usr/local/Cellar/cmake/3.17.3/share/cmake/Modules/Compiler/ARMClang.cmake:50 (message):
    CMAKE_SYSTEM_PROCESSOR must be set for ARMClang
    Supported processor: cortex-a5;cortex-a7;cortex-a8;cortex-a9;cortex-a12;cortex-a15;cortex-a17;cortex-m0;cortex-m0plus;cortex-m1;cortex-m3;cortex-m4;cortex-m7;cortex-m23;cortex-m33;cortex-m35p;cortex-r4;cortex-r4f;sc000;sc300
```

If I see the error above and I want o solve it, I will fix it by adding something to the ARMClang (see here toolchain) module.